OpenCeedlingFile.py

```diff
@@ -1,71 +1,84 @@
-import sublime
+from __future__ import print_function
+import os
+import re
+
 import sublime_plugin
-import re, inspect, os
+
 
 class OpenCeedlingFileCommand(sublime_plugin.WindowCommand):
+    def run(self, option):
+        if not self.window.active_view():
+            return
+
+        self.views = []
+        window = self.window
+        current_file_path = self.window.active_view().file_name()
+
+        if option == 'config':
+            config_matcher = re.compile(r"(\\\\)?(project|test)\.yml$")
+            self.open_project_file(config_matcher, self.window)
+
+        elif re.search(r"\w+\.[ch]\w*$", current_file_path):
+
+            current_file = re.search(r"([\w\.]+)$", current_file_path).group(1)
+            base_name = re.search(r"(\w+)\.(\w+)$", current_file).group(1)
+            base_name = re.sub('^test_', '', base_name)
+
+            print("Basename: " + base_name)
+
+            source_matcher = re.compile(r"[/\\\\]" + base_name + r"\.c$")
+            header_matcher = re.compile(r"[/\\\\]" + base_name + r"\.h$")
+            test_matcher = re.compile(r"[/\\\\]test_" + base_name + r"\.c$")

```

Comment:
Compiling these regular expressions  has absolutely no benefit if you're only using them exactly once later in the code.

Also, I know that `base_name` currently cannot contain any characters that have a special meaning in a regular expression outside of a single `.` (which matches everything), using `re.escape`on the string would be cleaner.

In fact, an even cleaner solution would be to perform simple string comparisons on the file names contained in the `files` list yielded from `os.walk` because none of the regular expressions look outside of a file's basename.